Brokerages/InteractiveBrokers/InteractiveBrokersSymbolMapper.cs

```diff
@@ -151,15 +151,21 @@ public Symbol GetLeanSymbol(string brokerageSymbol, SecurityType securityType, s
                         return Symbol.CreateOption(brokerageSymbol, market, OptionStyle.American, optionRight, strike, expirationDate);
 
                     case SecurityType.FutureOption:
-                        var canonicalFutureSymbol = Symbol.Create(GetLeanRootSymbol(brokerageSymbol), SecurityType.Future, market);
-                        var futureContractMonth = FuturesOptionsExpiryFunctions.GetFutureContractMonth(canonicalFutureSymbol, expirationDate);
-                        var futureExpiry = FuturesExpiryFunctions.FuturesExpiryFunction(canonicalFutureSymbol)(futureContractMonth);
+                        var future = FuturesOptionsUnderlyingMapper.GetUnderlyingFutureFromFutureOption(
+                            GetLeanRootSymbol(brokerageSymbol),
+                            market,
+                            expirationDate,
+                            DateTime.Now);
+
+                        if (future == null)
+                        {
+                            // This is the worst case scenario, because we didn't find a matching futures contract for the FOP.
+                            // Note that this only applies to CBOT symbols for now.
```

Comment:
This shouldn't happen now that the rules are in place right?

Reply:
It still can, if we request an expiry for a future that hasn't been listed (think: expiry super far ahead into the future) or does not have a FuturesListing rule whenever it hits the Serial contracts lookup in the FuturesOptionsUnderlyingMapper, though that should never happen.